mod-kb-ebsco-java/src/test/java/org/folio/KbEbscoApiTests.java

```diff
@@ -1,78 +1,105 @@
-    runFeature("classpath:common/destroy-data.feature");
-  }
-
-  @Test
-  void accessTypesTest() {
-    runFeatureTest("access-types");
-  }
-
-  @Test
-  void kbCredentialsTest() {
-    runFeatureTest("kb-credentials");
-  }
-
-  @Test
-  void packagesTest() {
-    runFeatureTest("packages");
-  }
-
-  @Test
-  void providersTest() {
-    runFeatureTest("providers");
-  }
-
-  @Test
-  void proxyTest() {
-    runFeatureTest("proxy");
-  }
-
-  @Test
-  void resourcesTest() {
-    runFeatureTest("resources");
-  }
-
-  @Test
-  void statusTest() {
-    runFeatureTest("status");
-  }
-
-  @Test
-  void tagsTest() {
-    runFeatureTest("tags");
-  }
-
-  @Test
-  void titlesTest() {
-    runFeatureTest("titles");
-  }
-
-  @Test
-  void userAssigmentTest() {
-    runFeatureTest("user-assignment");
-  }
+    private static final String TEST_BASE_PATH = "classpath:domain/mod-kb-ebsco-java/features/";
+    private static final String SETUP_CREDENTIALS_TAG = "CREDENTIALS";
+
+    public KbEbscoApiTests() {
+        super(new TestIntegrationService(new TestModuleConfiguration(TEST_BASE_PATH)));
+    }
+
+    @BeforeAll
+    public void setup() {
+        runFeature("classpath:domain/mod-kb-ebsco-java/kb-ebsco-junit.feature");
+    }
+
+    @BeforeEach
+    public void setupCredentials(TestInfo testInfo) {
+        if (testInfo.getTags().contains(SETUP_CREDENTIALS_TAG)) {
+            runFeature("classpath:domain/mod-kb-ebsco-java/features/setup/setup.feature");
+        }
+    }
+
+    @AfterAll
+    public void tearDown() {
+        runFeature("classpath:common/destroy-data.feature");
+    }
+
+    @AfterEach
+    public void destroyCredentials(TestInfo testInfo) {
+        if (testInfo.getTags().contains(SETUP_CREDENTIALS_TAG)) {
+            runFeature("classpath:domain/mod-kb-ebsco-java/features/setup/destroy.feature");
+        }
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void accessTypesTest() {
+        runFeatureTest("access-types");
+    }
+
+    @Test
+    void kbCredentialsTest() {
+        runFeatureTest("kb-credentials");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void packagesTest() {
+        runFeatureTest("packages");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void providersTest() {
+        runFeatureTest("providers");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void proxyTest() {
+        runFeatureTest("proxy");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void resourcesTest() {
+        runFeatureTest("resources");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void statusTest() {
+        runFeatureTest("status");
+    }
+
+    @Test
+    void tagsTest() {
+        runFeatureTest("tags");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void titlesTest() {
+        runFeatureTest("titles");
+    }
+
+    @Test
+    @Tag(SETUP_CREDENTIALS_TAG)
+    void userAssigmentTest() {
+        runFeatureTest("user-assignment");
+    }
```

Comment:
Something went wrong with code style. To many intentions (should be 2 spaces).

Reply:
Thanks for noticing 👍